aion-api/src/main/java/com/aion/dashboard/controllers/Dashboardv2.java

```diff
@@ -667,4 +651,12 @@ boolean isNotEmpty(String str){
             blockNumber.map(statisticsService::transactionStats).orElseGet(statisticsService::transactionStats)
         ));
     }
+
+    @GetMapping(value = "/circulatingSupply")
+    public ResponseEntity<Result<CirculatingSupplyDTO>> getCirculatingSupply(){
+        return packageResponse(CirculatingSupplyMapper
+                .getMapper()
+                .makeResult(circulatingSupplyJPARepository.findAll().get(0)));
```

Comment:
I'm not sure about how efficienct this is under the hood (whether the query is lazy or not) but it seems like you could make this more efficient by indexing on timestamp, and querying by the latest entry from the last day or so. Also is there caching for this endpoint?

Reply:
Index, cache, and limiting query  was added